pkg/queue/cluster_queue.go

```diff
@@ -89,17 +91,27 @@ func workloadKey(i *workload.Info) string {
 	return workload.Key(i.Obj)
 }
 
-func newClusterQueue(cq *kueue.ClusterQueue, wo workload.Ordering) (*ClusterQueue, error) {
-	cqImpl := newClusterQueueImpl(wo, realClock)
+func newClusterQueue(ctx context.Context, client client.Client, cq *kueue.ClusterQueue, wo workload.Ordering, fsConfig *config.AdmissionFairSharing) (*ClusterQueue, error) {
+	enableAdmissionFs, fsResWeights := isAdmissionFsEnabled(cq, fsConfig)
+	cqImpl := newClusterQueueImpl(ctx, client, wo, realClock, fsResWeights, enableAdmissionFs)
 	err := cqImpl.Update(cq)
 	if err != nil {
 		return nil, err
 	}
 	return cqImpl, nil
 }
 
-func newClusterQueueImpl(wo workload.Ordering, clock clock.Clock) *ClusterQueue {
-	lessFunc := queueOrderingFunc(wo)
+func isAdmissionFsEnabled(cq *kueue.ClusterQueue, fsConfig *config.AdmissionFairSharing) (bool, map[corev1.ResourceName]float64) {
```

Comment:
```suggestion
func afsResourceWeights(cq *kueue.ClusterQueue, fsConfig *config.AdmissionFairSharing) (bool, map[corev1.ResourceName]float64) {
```
if is a bit weird to call function "is" when returning more than bool. The bool probably should be the second argument, as more idiomatic (as golang returns "ok" bool as second when checking maps).

Reply:
Ditto

Reply:
> The bool probably should be the second argument, as more idiomatic (as golang returns "ok" bool as second when checking maps).

this is not addressed I think

Reply:
Can you double check if you looking at the latest changes?

Reply:
here: https://github.com/kubernetes-sigs/kueue/pull/4687/files#diff-996d1d16eaefed60407545849893ea41fac4b85c04b78bb396bb93fcf6004bc8R104

Reply:
![image](https://github.com/user-attachments/assets/012018d5-ce2f-42d8-8d74-c2dfdf5e6673)